### PR TITLE
add vue support

### PR DESCRIPTION
**Pull request by tjx666**

### CODE REVIEW

**Comment by tjx666** [comment]
this pr should be merged after another.

**Comment by tjx666** [comment]
I get cold, and need to sleep...

**Comment by daidodo** [comment]
Thanks for the PR!

Could you resolve conflicts and update the base version?

**Comment by tjx666** [comment]
I update version to 4.0.0, because we only support node14+ now.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "format-imports",
-  "version": "3.1.5",
+  "version": "4.0.0",
```
Comment (daidodo) [inline package.json:3]: I didn't mean to update this version. Please revert it and I'll update it if needed.

Comment (daidodo) [inline package.json:3]: Sorry for the confusion! I didn't mean to update this version. Please revert it and I'll update it if needed.

File: src/lib/format/main/index.ts [hunk 2]
```diff
@@ -13,12 +15,19 @@ import { formatSource } from './format';
...
 /**
  * File extensions supported.
  */
-export const SUPPORTED_EXTENSIONS: Extension[] = ['js', 'ts', 'jsx', 'tsx'];
+export const SUPPORTED_EXTENSIONS: Extension[] = ['js', 'ts', 'jsx', 'tsx', 'vue'];
+
+const CRLF = '\r\n';
+const LF = '\n';
+const CR = '\r';
+function getLineEndingAfterVueScriptTag(text: string) {
```
Comment (daidodo) [inline src/lib/format/main/index.ts:28]: Please reuse https://github.com/daidodo/format-imports/blob/main/package.json#L150

File: src/lib/format/main/index.ts [hunk 3]
```diff
@@ -48,6 +57,30 @@ export async function formatSourceFromFile(
   log.debug('Config:', config);
   log.debug('Options:', options);
   const allConfig = await enhanceConfig(config, fileName, options);
+
+  if (fileName.endsWith('.vue')) {
+    const { descriptor } = parseVue(text);
+    const vueScript = descriptor.scriptSetup ?? descriptor.script;
```
Comment (daidodo) [inline src/lib/format/main/index.ts:63]: Is it true that there is ONLY one script in a Vue source?

**Comment by daidodo** [comment]
@tjx666 
Really appreciate your PR as it extends the plugin!

I can't wait to merge it as soon as the main comments are addressed.

File: src/lib/format/main/index.ts (hunk 3 above)
Comment (tjx666) [reply src/lib/format/main/index.ts:63]: https://github.com/vuejs/vue-loader/issues/228

**Review by daidodo — APPROVED** [review]

**Pull request merged**